### PR TITLE
Adds Settings Dialog

### LINKED ISSUE #20: Create settings dialog
Create a dialog for changing settings. Depends on #6.

Note that the `Settings` object is a Pydantic model, so we can use the `ValidatedInputWidget` in #16 to automatically create the relevant input widget for different data types.

I would recommend creating a local 'draft' `Settings` object for changing settings in the widget, which is then saved to `self.view.settings` when saved. This makes it easier to reset unsaved settings to project or global settings (for global, just assign a default `Settings()` to `self.view.settings`) without increasing the size of the `Settings` object's interface.

Acceptance:
- settings widget accessible via toolbar
- different tabs for different setting groups
- way to save settings, reset settings to either project settings or global defaults

### PR DESCRIPTION
**Pull request by DrPaulSharp**
Closes #20 

This PR introduces a settings dialog, with tabs mapped to the settings groups established in the settings model. Project settings are updated with any changes made, excluding resetting to the global default.

This PR also addresses a bug in the code in that menu actions affecting a project were all enabled (unlike the taskbar which was correctly disabled), causing the GUI to crash in some cases. The appropriate menu actions have been disabled on startup.

### CODE REVIEW

File: rascal2/dialogs/settings_dialog.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        Parameters
+        ----------
+        parent : MainWindowView
+            The view of the RasCAL-2 GUI
+        """
+        super().__init__(parent)
+
+        self.setModal(True)
+        self.setMinimumWidth(400)
```
Comment (StephenNneji) [inline rascal2/dialogs/settings_dialog.py:22]: I think you can go a bit bigger on the min width but also set a min height. something around w: 600, h: 400 looked good to me

File: rascal2/dialogs/settings_dialog.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        self.setWindowTitle("Settings")
+
+    def update_settings(self) -> None:
+        """Accept the changed settings"""
+        self.parent().settings = self.settings.copy()
+        self.parent().settings.save(self.parent().save_path)
+        self.accept()
+
+    def reset_default_settings(self) -> None:
+        """Reset the settings to the global defaults"""
```
Comment (StephenNneji) [inline rascal2/dialogs/settings_dialog.py:57]: The _setting.json_ file needs to be removed after a reset. maybe add a clear method in setting to delete it

**Review by StephenNneji — COMMENTED** [review]
Looks good, please see comments

File: rascal2/ui/view.py [hunk 3]
```diff
@@ -225,6 +270,7 @@ def init_settings_and_log(self, save_path: str):
             The save path for the project.
 
         """
+        self.save_path = save_path
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply rascal2/ui/view.py:273]: I tried that, and it causes the GUI to crash for some reason. Any idea why?

File: rascal2/dialogs/settings_dialog.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        for i, setting in enumerate(group_settings):
+            label_text = setting.replace("_", " ")
+            tab_layout.addWidget(QtWidgets.QLabel(label_text), i, 0)
+            self.widgets[setting] = ValidatedInputWidget(field_info[setting])
+            try:
+                self.widgets[setting].set_data(getattr(self.settings, setting))
+            except TypeError:
+                self.widgets[setting].set_data(str(getattr(self.settings, setting)))
+            self.widgets[setting].edited_signal.connect(self.create_slot(setting))
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply rascal2/dialogs/settings_dialog.py:91]: If I try to use a lambda here, only modify the last setting in a particular tab can be modified. Changing the values of others does not communicate to the settings object. Is there anything I've missed to make this work? My code is:

```
            self.widgets[setting].edited_signal.connect(lambda: self.modify_setting(setting))
            tab_layout.addWidget(self.widgets[setting], i, 1)

        tab_layout.setAlignment(QtCore.Qt.AlignmentFlag.AlignTop)
        self.setLayout(tab_layout)

    def modify_setting(self, setting: str):
        setattr(self.settings, setting, self.widgets[setting].get_data())
```

File: rascal2/ui/view.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply rascal2/ui/view.py:273]: The test crashes, the code itself is fine.

File: rascal2/dialogs/settings_dialog.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        self.setModal(True)
+        self.setMinimumWidth(400)
+
+        self.settings = parent.settings.copy()
+        self.reset_dialog = None
+
+        tab_widget = QtWidgets.QTabWidget()
+        tab_widget.addTab(SettingsTab(self, SettingsGroups.General), SettingsGroups.General)
+        tab_widget.addTab(SettingsTab(self, SettingsGroups.Logging), SettingsGroups.Logging)
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply rascal2/dialogs/settings_dialog.py:32]: Let's take the logging tab out for now at least then. It definitely seems odd to change where logs are saved, but have the settings fixed.

**Review by StephenNneji — APPROVED** [review]

File: rascal2/core/settings.py [hunk 6]
```diff
@@ -22,6 +22,26 @@ def get_global_settings() -> QtCore.QSettings:
     )
 
 
+def delete_local_settings(path: str | PathLike) -> None:
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply rascal2/core/settings.py:25]: Hmm, I was thinking about this, but does the settings file correspond with a particular instance of the Settings object? (The save routine is in the object as well though so . . .)

File: tests/test_dialogs.py [hunk 7]
```diff
@@ -1,12 +1,20 @@
...
-from PyQt6 import QtWidgets
+from PyQt6 import QtCore, QtWidgets
 
+from rascal2.core.settings import LogLevels, Settings, SettingsGroups
 from rascal2.dialogs.project_dialog import ProjectDialog
+from rascal2.dialogs.settings_dialog import SettingsDialog, SettingsTab
+from rascal2.widgets.inputs import ValidatedInputWidget
 
 
 class MockPresenter(QtWidgets.QMainWindow):
```
[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply tests/test_dialogs.py:12]: I've changed this and everything seems fine.

**Pull request merged**